Reflection/ReflectionClass.php

```diff
@@ -0,0 +1,669 @@
+     * @since 5.4
+     */
+    public function isCloneable(): bool
+    {
+    }
+
+    /**
+     * Gets the filename of the file in which the class has been defined
+     *
+     * @link https://php.net/manual/en/reflectionclass.getfilename.php
+     * @return string|false the filename of the file in which the class has been defined.
+     * If the class is defined in the PHP core or in a PHP extension, {@see false}
+     * is returned.
+     */
+    public function getFileName(): string|false
+    {
+    }
+
+    /**
+     * Gets starting line number
+     *
+     * @link https://php.net/manual/en/reflectionclass.getstartline.php
+     * @return int The starting line number, as an integer.
+     */
+    public function getStartLine(): int
+    {
+    }
+
+    /**
+     * Gets end line
+     *
+     * @link https://php.net/manual/en/reflectionclass.getendline.php
+     * @return int|false The ending line number of the user defined class, or
+     * {@see false} if unknown.
+     */
+    public function getEndLine(): int|false
+    {
+    }
+
+    /**
+     * Gets doc comments
+     *
+     * @link https://php.net/manual/en/reflectionclass.getdoccomment.php
+     * @return string|false The doc comment if it exists, otherwise {@see false}
+     */
+    public function getDocComment(): string|false
+    {
+    }
+
+    /**
+     * Gets the constructor of the class
+     *
+     * @link https://php.net/manual/en/reflectionclass.getconstructor.php
+     * @return ReflectionMethod|null A {@see ReflectionMethod} object reflecting
+     * the class' constructor, or {@see null} if the class has no constructor.
+     */
+    public function getConstructor(): ?ReflectionMethod
+    {
+    }
+
+    /**
+     * Checks if method is defined
+     *
+     * @link https://php.net/manual/en/reflectionclass.hasmethod.php
+     * @param string $name Name of the method being checked for.
+     * @return bool Returns {@see true} if it has the method, otherwise {@see false}
+     */
+    public function hasMethod(string $name): bool
+    {
+    }
+
+    /**
+     * Gets a <b>ReflectionMethod</b> for a class method.
+     *
+     * @link https://php.net/manual/en/reflectionclass.getmethod.php
+     * @param string $name The method name to reflect.
+     * @return ReflectionMethod A {@see ReflectionMethod}
+     * @throws \ReflectionException if the method does not exist.
+     */
+    public function getMethod(string $name): ReflectionMethod
+    {
+    }
+
+    /**
+     * Gets an array of methods for the class.
+     *
+     * @link https://php.net/manual/en/reflectionclass.getmethods.php
+     * @param int $filter Filter the results to include only methods
+     * with certain attributes. Defaults to no filtering.
+     * @return ReflectionMethod[] An array of {@see ReflectionMethod} objects
+     * reflecting each method.
+     */
+    public function getMethods(int $filter = 119): array
```

Comment:
What does this magic number mean? Also doc says that "Defaults to no filtering." so seems it should be null

Reply:
> What does this magic number mean

This is a bitmask of all available values: `RefelctionMethod::IS_STATIC | RefelctionMethod::IS_PUBLIC | etc...`

Reply:
Yes, indeed, the argument may be null. My bad)

Reply:
Fixed: https://github.com/JetBrains/phpstorm-stubs/pull/857/commits/0e9a2fcdfa3dfe65f18a4cb69d42475245a1d05a